flow/render_status.py

```diff
@@ -0,0 +1,220 @@
+# make separate python class for render_status
+from .util import mistune
+from .scheduling.base import JobStatus
+
+
+class renderer:
+    """A class for rendering status in different format.
+
+    This class provides method and string output for rendering status output in different format,
+    currently supports: terminal, markdown and html
+
+    """
+
+    def __init__(self):
+        self.markdown_output = None
+        self.terminal_output = None
+        self.html_output = None
+
+    def generate_terminal_output(self):
+        self.terminal_output = mistune.terminal(self.markdown_output)
+
+    def generate_html_output(self):
+        self.html_output = mistune.html(self.markdown_output)
+
+    def render(self, template, template_environment, context, file, detailed, expand,
+               unroll, compact, pretty, option):
+        """Render the status in different format for print_status.
+
+        :param template:
+            User provided Jinja2 template file.
+        :type template:
+            str
+        :param template_environment:
+            template environment
+        :type template_environment:
+            :class:'FlowProject._template_environment'
+        :param context:
+            Context that includes all the information for rendering status output.
+        :type context:
+            dict
+        :param file:
+            Redirect all output to this file, defaults to sys.stdout.
+        :type file:
+            str
```

Comment:
Is this supposed to be a `str` or something else? A file-like object? Not sure what the right word is.